### PR TITLE
fix(maven): don't cache private

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Restricts maven caching for public packages only.

## Context:

We shouldn't cache private packages until we can be properly support it.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/maven/index.ts [hunk 1]
```diff
@@ -37,21 +37,27 @@ function getMavenUrl(
...
 async function downloadMavenXml(
   pkgUrl: url.URL | null
-): Promise<XmlDocument | null> {
+): Promise<MavenXml | null> {
   /* istanbul ignore if */
   if (!pkgUrl) {
     return null;
   }
   let rawContent: string;
+  let authorization: boolean;
```
Comment (viceice) [inline lib/datasource/maven/index.ts:53]: Not yet used?

**Review by viceice — COMMENTED** [review]
Needs coverage fix

File: lib/datasource/maven/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/datasource/maven/index.ts:53]: fixed

**Review by viceice — COMMENTED** [review]
Small test thing, otherwise LGTM

**Review by viceice — APPROVED** [review]

**Pull request merged**